lib/remote-model.js

```diff
@@ -166,21 +143,63 @@ class RemoteModel extends LocalModel {
         return _.extend({ status: statusCode }, data);
     }
 
+    url() {
+        return;
+    }
 
-    url(options) {
-        options = options || {};
-        let opts = {};
-        if (options.url) {
-            opts = url.parse(options.url);
+    auth(credentials) {
+        if (!credentials) return;
+
+        if (typeof credentials === 'string') {
+            let auth = credentials.split(':');
+            credentials = {
+                user: auth.shift(),
+                pass: auth.join(':'),
+                sendImmediately: true
+            };
         }
-        // passing a host to url.format overrides other options, so remove it
-        delete opts.host;
-        _.extend(opts, options);
-        return url.format(opts);
+
+        return credentials;
```

Comment:
is the intention to call this as `auth() { return super.auth('username:password'); }` ?

Reply:
Yes, and if needed a subclass can create its own object that matches the expected format.